src/helpers/get-items-from-portal.js

```diff
@@ -0,0 +1,50 @@
+const axios = require('axios');
+const { serializeQueryParams } = require('./portal-query-builder');
+
+async function getPortalItems(portalUrl, query, maxPageSize) {
+    const firstPage = await fetchItemsFromPortal(portalUrl, query);
+    const totalBatch = getTotalBatch(firstPage.total, maxPageSize);
+    if (totalBatch === 1) {
+        return {
+            items: firstPage.results,
+            count: firstPage.total
+        };
+    }
+    const remainingRequests = buildRemainingPageRequests({ portalUrl, query, totalBatch, maxPageSize });
+    const remainingItems = await getRemainingPortalItems(remainingRequests);
+    return {
+        items: [...firstPage.results, ...remainingItems],
+        count: firstPage.total
+    };
+};
+
+async function fetchItemsFromPortal(portalUrl, query) {
+    const url = `${portalUrl}?${serializeQueryParams(query)}`;
+    const items = await axios.get(url);
+    return items.data;
+}
+
+function getTotalBatch(total, maxPageSize) {
+    return Math.ceil(total / maxPageSize);
+}
+
+function buildRemainingPageRequests({ portalUrl, query, totalBatch, maxPageSize }) {
+    const requests = [];
+    const portalQuery = Object.assign({}, query);
+    // Multiple batches each with maxPageSize number of records
+    for (let i = 1; i < totalBatch; i++) {
+        portalQuery.start += maxPageSize;
+        requests.push(axios.get(`${portalUrl}?${serializeQueryParams(portalQuery)}`));
```

Comment:
ok, so the reason you don't have a mutation problem with `start` is that you serialize the query params within each iteration of the loop.  If you are going to do that, you don't need to clone.